xonsh/xontribs_meta.py

```diff
@@ -0,0 +1,639 @@
+import functools
+import typing as tp
+
+
+class XontribPkg(tp.NamedTuple):
+    install: tp.Dict[str, str]
+    license: str = ""
+    name: str = ""
+    url: tp.Optional[str] = None
+
+
+class Xontrib(tp.NamedTuple):
+    """Meta class that is used to descript xontribs.
+
+    Attributes
+    ----------
+        url: str
+            url
+        description: str
+            short description about the xontrib.
+        package: XontribPkg
+            pkg information for installing the xontrib
+        tags: tuple of str
```

Comment:
This block should not be indented, and there should be a colon between the name and the type, eg.

```
    Attributes
    ----------
    url : str
        url
    description : str
```